app/lib/actions/pulp3/repository/sync.rb

```diff
@@ -10,7 +10,9 @@ def plan(repo, smart_proxy, options = {})
 
         def invoke_external_task
           repo = ::Katello::Repository.find(input[:repo_id])
-          output[:pulp_tasks] = repo.backend_service(::SmartProxy.find(input[:smart_proxy_id])).sync
+          mirror_param = input.dig(:options, :options, :remove_missing)
+          mirror = mirror_param.nil? ? true : mirror_param
```

Comment:
why are we not looking on the repository to see if mirror_on_sync is set?

Reply:
also, this should be set in the importer, not at sync time: https://github.com/Katello/katello/blob/master/app/services/katello/pulp/repository/yum.rb#L8

Reply:
Updated this to use mirror_on_sync field and send it to remote with sync call.